R/module_teal_data.R

```diff
@@ -96,16 +96,17 @@ srv_validate_reactive_teal_data <- function(id, # nolint: object_length
                                             data,
                                             modules = NULL,
                                             validate_shiny_silent_error = FALSE) {
-  moduleServer(id, function(input, output, session) {
-    if (!is.reactive(data)) {
-      stop("The `teal_data_module` passed to `data` must return a reactive expression.", call. = FALSE)
-    }
+  checkmate::assert_string(id)
+  assert_reactiveExpr(data)
```

Comment:
Imagine you're app developer and you get an error message:

```
Error in srv_validate_reactive_teal_data(id = "validate", data = data_out,  : 
  Assertion on 'data' failed: Must be a reactive expression (i.e. inherit from 'reactiveExpr' class) but has class 'character'.
```

While on main it is:

```
The `teal_data_module` passed to `data` must return a reactive expression.
```

----

The message should be directed to app developer not to us. App developer should not know anything about `srv_validate_teal_data`, this is why `call. = FALSE` was needed.